Mathlib/RingTheory/LittleWedderburn.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Eric Rodriguez
+-/
+import Mathlib.GroupTheory.ClassEquation
+import Mathlib.GroupTheory.GroupAction.ConjAct
+import Mathlib.RingTheory.Polynomial.Cyclotomic.Eval
+
+/-!
+# Wedderburn's Little Theorem
+
+This file proves Wedderburn's Little Theorem.
+
+## Main Declarations
+
+* `littleWedderburn`: a finite division ring is a field.
+
+## Future work
+If interested, generalisations to semifields could be explored. The theory of semi-vector spaces is
+not clear, but assuming that such a theory could be found where every module considered in the
+below proof is free, then the proof works nearly verbatim.
+
+-/
+
+
+open scoped BigOperators Polynomial
+
+/- Everything in this namespace is internal to the proof of Wedderburn's little theorem. -/
+
+namespace LittleWedderburn
+
+variable (D : Type*) [DivisionRing D]
+
+private def InductionHyp : Prop :=
+  ∀ R : Subring D, R < ⊤ → ∀ ⦃x y⦄, x ∈ R → y ∈ R → x * y = y * x
+
+namespace InductionHyp
+
+open FiniteDimensional Polynomial
+
+variable {D}
+
+private def field (hD : InductionHyp D) {R : Subring D} (hR : R < ⊤)
+  [Fintype D] [DecidableEq D] [DecidablePred (· ∈ R)] :
+    Field R :=
+  { show DivisionRing R from Fintype.divisionRingOfIsDomain R with
+    mul_comm := fun x y ↦ Subtype.ext <| hD R hR x.2 y.2 }
+
+-- this proof is currently hard to read through because of `(config := {zeta := false})` expanding
+-- all local definitions.
+private theorem center_eq_top [Finite D] (hD : InductionHyp D) : Subring.center D = ⊤ := by
+  classical
```

Comment:
Could you sketch in a comment the proof you are doing, both here and in the file-level docstring?